File: src/PhantasyIslandPythonRemoteControl/radio/api_fly.py
"""
场景相关 API 模块。
"""

from __future__ import annotations

import typing

from .api_module import ApiModule


class FlyApi(ApiModule):

    def listFlyObject(self):
        return self.send('fly.listFlyObject')

    def getFlyObjectInfo(self, keyName: str):
        return self.send('fly.getFlyObjectInfo', data={'keyName': keyName})

    def getFlyObjectCameraImageDown(self, keyName: str):
        return self.send('fly.getFlyObjectCameraImageDown', data={'keyName': keyName})

    def getFlyObjectCameraImageFront(self, keyName: str):
        return self.send('fly.getFlyObjectCameraImageFront', data={'keyName': keyName})

File: src/rb/PhantasyIslandRubyRemoteControl/radio/api_fly.rb
require_relative 'api_module'

class FlyApi < ApiModule
  def listFlyObject
    send('fly.listFlyObject')
  end

  def getFlyObjectInfo(keyName)
    send('fly.getFlyObjectInfo', { 'keyName' => keyName })
  end

  def getFlyObjectCameraImageDown(keyName)
    send('fly.getFlyObjectCameraImageDown', { 'keyName' => keyName })
  end

  def getFlyObjectCameraImageFront(keyName)
    send('fly.getFlyObjectCameraImageFront', { 'keyName' => keyName })
  end
end
